consensus/XDPoS/engines/engine_v2/engine.go

```diff
@@ -930,6 +930,10 @@ func (x *XDPoS_v2) commitBlocks(blockChainReader consensus.ChainReader, proposed
 			Round:  round,
 		}
 		log.Debug("Successfully committed block", "Committed block Hash", x.highestCommitBlock.Hash, "Committed round", x.highestCommitBlock.Round)
+		// Perform forensics related operation
+		var headerQcToBeCommitted []types.Header
+		headerQcToBeCommitted = append(headerQcToBeCommitted, *parentBlock, *proposedBlockHeader)
+		go x.forensics.SetCommittedQCs(headerQcToBeCommitted, *incomingQc)
```

Comment:
don't understand why you use array to send the parentBlock and proposedBlockheader, the order matters in your implementation.

Reply:
Correct. Array is more flexible than passing parameters. I also have verification in place to check the order is correct in the array.